network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/exclusion/ui/ManuallyDisableAppProtectionDialog.kt

```diff
@@ -0,0 +1,131 @@
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class ManuallyDisableAppProtectionDialog : DialogFragment() {
+
+    interface ManuallyDisableAppProtectionDialogListener {
+        fun onAppProtectionDisabled(
+            appName: String,
+            packageName: String,
+            report: Boolean = false,
+        )
+    }
+
+    val listener: ManuallyDisableAppProtectionDialogListener
+        get() {
+            return if (parentFragment is ManuallyDisableAppProtectionDialogListener) {
+                parentFragment as ManuallyDisableAppProtectionDialogListener
+            } else {
+                activity as ManuallyDisableAppProtectionDialogListener
+            }
+        }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        isCancelable = false
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val rootView = layoutInflater.inflate(R.layout.dialog_netp_manually_disable_app, null)
+
+        val appIcon = rootView.findViewById<ImageView>(R.id.netp_disable_dialog_app_icon)
+        val appLabel = rootView.findViewById<TextView>(R.id.netp_disable_dialog_label)
+        val reportCTA = rootView.findViewById<Button>(R.id.netp_disable_dialog_action_report)
+        val skipCTA = rootView.findViewById<Button>(R.id.netp_disable_dialog_action_skip)
+
+        val alertDialog = MaterialAlertDialogBuilder(requireActivity(), com.duckduckgo.mobile.android.R.style.Widget_DuckDuckGo_Dialog)
+            .setView(rootView)
+
+        validateBundleArguments()
+        isCancelable = false
+
+        populateAppIcon(appIcon)
+        populateAppName(appLabel)
+        configureListeners(reportCTA, skipCTA)
+
+        return alertDialog.create()
+    }
+
+    private fun populateAppIcon(appIcon: ImageView) {
+        val icon = requireActivity().packageManager.safeGetApplicationIcon(getPackageName())
+        appIcon.setImageDrawable(icon)
+    }
+
+    private fun populateAppName(appLabel: TextView) {
+        appLabel.text = getString(R.string.netpExclusionListManuallyDisableAppDialogLabel, getAppName())
+    }
+
+    private fun getPackageName(): String {
+        return requireArguments().getString(KEY_APP_PACKAGE_NAME)!!
+    }
+
+    private fun getAppName(): String {
+        return requireArguments().getString(KEY_APP_NAME)!!
+    }
+
+    private fun configureListeners(
+        reportCTA: Button,
+        skipCTA: Button,
+    ) {
+        reportCTA.setOnClickListener {
+            dismiss()
+            listener.onAppProtectionDisabled(getAppName(), getPackageName(), true)
+        }
+
+        skipCTA.setOnClickListener {
+            dismiss()
+            listener.onAppProtectionDisabled(getAppName(), getPackageName())
+        }
+    }
+
+    private fun validateBundleArguments() {
+        if (arguments == null) throw IllegalArgumentException("Missing arguments bundle")
```

Comment:
`requireArguments` below already will fire exception in case of arguments being null, so this line is redundant.